File: ProjetASI2/PersonalFinanceManagementAppFrontEnd/src/app/Model/Person.ts


export class Person {
    id: number | null;
    firstname: string;
    lastName: string;
    email: string;
    totalBudget: number;
    totalBudgetSpend: number;
    currenttotalBudget: number;


    constructor(
        id: number | null,
        firstname: string,
        lastName: string,
        email: string,
        totalBudget: number,
        totalBudgetSpend: number,
        currenttotalBudget: number
    ) {
        this.id = id;
        this.firstname = firstname;
        this.lastName = lastName;
        this.email = email;
        this.totalBudget = totalBudget;
        this.totalBudgetSpend = totalBudgetSpend;
        this.currenttotalBudget = currenttotalBudget;
    }
}

File: ProjetASI2/Personal-Finance-Management-Spring-boot/src/main/java/com/PersonalFinanceManagementApplication/Personal/Finance/Management/Application/Model/Person.java
package com.PersonalFinanceManagementApplication.Personal.Finance.Management.Application.Model;


import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.util.Set;

@Entity
@Table(name = "Persons")
@Getter
@Setter
@ToString
@AllArgsConstructor
@NoArgsConstructor
public class Person {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(
            nullable = false,
            columnDefinition = "TEXT"
    )
    private String firstname;

    @Column(
            nullable = false,
            columnDefinition = "TEXT"
    )
    private String lastName;

    @Column(
            nullable = false,
            columnDefinition = "TEXT",
            unique = true
    )
    private String email;

    private BigDecimal totalBudget = BigDecimal.ZERO;

    private BigDecimal totalBudgetSpend = BigDecimal.ZERO;

    private BigDecimal currenttotalBudget = BigDecimal.ZERO;

}
